src/nvim/tui/tui.c

```diff
@@ -69,6 +70,9 @@ typedef struct {
   bool can_use_terminal_scroll;
   bool mouse_enabled;
   bool busy;
+  /* use int instead ? */
+  // cursorentry_T cursor_shapes[SHAPE_IDX_COUNT]; /**!< is it fine to reuse cursorentry_T here ? */
+  int cursor_shapes[SHAPE_IDX_COUNT]; /**!< is it fine to reuse cursorentry_T here ? */
```

Comment:
Of course, the conversion happened once the TUI received all data. Looks like it complexifies things instead of making them simpler so forget about it xD